egs/librispeech/ASR/conformer_ctc/conformer.py

```diff
@@ -193,6 +194,18 @@ def __init__(
 
         self.normalize_before = normalize_before
 
+        self.kernel_size = 21
```

Comment:
All these details should probably be in a module, e.g. Conv1dAbsModule, that will hold all this logic.
I think it's better if you get practice writing this kind of thing.

Reply:
OK, will pay attention it in the future.

Reply:
Try keeping the multi-head attention but adding this after the multi-head attention.
(i.e. as a separate block, so instead of feeforward+mha+conv+macaron, you have feedforward+mha+conv1dabs+conv+macaron.

Reply:
OK, will try it based on kernel_size=21.

Reply:
ok.  kernel_size=21 is still quite small, I would try something like 31 or 63.
(Note: these are power-of-2 minus one, which will tend to be efficient for implementation).

Reply:
Oh, seems better than 21. Will try 31 or 63 if the training process is normal when kernel_size is 21.

Reply:
MHA can grasp the global information for some step (frame). And Conv1d focuses on the local information for some step (frame). Hope to this can help accelerate the training process.